src/main/java/seedu/pill/util/TransactionManager.java

```diff
@@ -49,36 +49,17 @@ public TransactionManager(ItemMap itemMap) {
      * @throws PillException - If there's insufficient stock for an outgoing transaction or if any other validation
      *                       fails
      */
-    public Transaction createTransaction(String itemName, int quantity,
+    public Transaction createTransaction(String itemName, int quantity, LocalDate expiryDate,
                                          Transaction.TransactionType type,
                                          String notes, Order associatedOrder) throws PillException {
 
         Transaction transaction = new Transaction(itemName, quantity, type, notes, associatedOrder);
 
         if (type == Transaction.TransactionType.INCOMING) {
-            Item item = new Item(itemName, quantity);
+            Item item = new Item(itemName, quantity, expiryDate);
             itemMap.addItem(item);
         } else {
-            TreeSet<Item> items = itemMap.get(itemName);
-            int totalQuantity = items.stream()
-                    .mapToInt(Item::getQuantity)
-                    .sum();
-
-            if (totalQuantity < quantity) {
-                throw new PillException(ExceptionMessages.INVALID_QUANTITY);
-            }
-
-            int remainingQty = quantity;
-            while (remainingQty > 0) {
-                Item item = items.first();
-                if (item.getQuantity() <= remainingQty) {
-                    remainingQty -= item.getQuantity();
-                    itemMap.deleteItem(itemName, item.getExpiryDate());
-                } else {
-                    item.setQuantity(item.getQuantity() - remainingQty);
-                    remainingQty = 0;
-                }
-            }
+            itemMap.useItem(itemName, quantity);
```

Comment:
Excellent use of abstraction and removing code repetition. However, the use item command consumes the earliest expiring items and not the items with the expiry specified in the order. Please look into this